### PR TITLE
Add initial handling for content flagging feature

### PR DESCRIPTION
**Pull request by philcable**
- Introduce support for flagging when a comment is about bad content.
- Add a "Flagged Posts" dashboard page that repurposes the comments list table.
- Remove the `Unapprove` and `Approve` comment row actions.
- Renames the `Reply` row action as `Resolve`.
- Add a checkbox for resolving a flagged comment to the inline `Reply` form, checked by default.
- Enqueue the comment reply script if appropriate.

This assumes that:
- the Knowledge Base post type needs to support regular comments as well as comments for flagging bad content;
- the comment form on Knowledge Base posts should display only for authorized users, regardless of Discussion settings;
- the approval process is not applicable to comments on Knowledge Base posts;
- a note should be captured whenever a comment about bad content is marked as resolved;
- there may be cases where a reply to a comment about bad content is not meant to resolve the concern (in which case the user could uncheck the box);
- spam will not be an issue for comments on Knowledge Base posts (possibly a false assumption);
- "Comments" is an okay heading for the "Flagged Posts" dashboard page :).

This should be rebased against master after https://github.com/happyprime/wsu-sfs411/pull/2 is merged.

### CODE REVIEW

File: articles/post-knowledge_base.php [hunk 1]
```diff
@@ -85,7 +85,12 @@
 		</div>
 	<?php endif; ?>
 
-	<?php comments_template(); ?>
+	<?php
+	// Display the comments template if the current user is logged in.
+	//if ( is_user_logged_in() ) {
```
Comment (jeremyfelt) [inline articles/post-knowledge_base.php:90]: Indecisive? :trollface:

File: includes/flagged-posts.php [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+		}
+
+		addResolveCheckbox();
+		removeFilterActions();
+	</script>
+	<?php
+}
+
+/**
+ * If the comment is about bad content, save it as a `flagged_content` type.
```
Comment (jeremyfelt) [inline includes/flagged-posts.php]: I _think_ this comment needs to be updated to reflect the "resolved" comment type.

File: includes/flagged-posts.php [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+		'post_type' => 'knowledge_base',
+		'count'     => true,
+	);
+
+	// Add the argument for trashed comments.
+	$trashed_comments_args = array_merge(
+		$args,
+		array(
+			'status' => 'trash',
+		),
```
Comment (jeremyfelt) [inline includes/flagged-posts.php:259]: Pretty sure this is already fixed in another PR :)

**Review by jeremyfelt — APPROVED** [review]
👍 in general, though I left a couple nit-picky comments :)

File: articles/post-knowledge_base.php (hunk 1 above)
Comment (philcable) [reply articles/post-knowledge_base.php:90]: Haha, yep! Reached a decision after discussing with @slocker3 and addressed this in another PR 🙂

File: includes/flagged-posts.php (hunk 2 above)
Comment (philcable) [reply includes/flagged-posts.php]: It does indeed!

**Pull request merged**